superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx

```diff
@@ -0,0 +1,1307 @@
+  const checkNotificationSettings = () => {
+    if (!notificationSettings.length) {
+      return false;
+    }
+
+    let hasInfo = false;
+
+    notificationSettings.forEach(setting => {
+      if (!!setting.method && setting.recipients?.length) {
+        hasInfo = true;
+      }
+    });
+
+    return hasInfo;
+  };
+
+  const validate = () => {
+    if (
+      currentAlert &&
+      currentAlert.name?.length &&
+      currentAlert.owners?.length &&
+      currentAlert.crontab?.length &&
+      ((contentType === 'dashboard' && !!currentAlert.dashboard) ||
+        (contentType === 'chart' && !!currentAlert.chart)) &&
+      checkNotificationSettings()
+    ) {
+      if (isReport) {
+        setDisableSave(false);
+      } else if (
+        !!currentAlert.database &&
+        currentAlert.sql?.length &&
+        !!currentAlert.validator_config_json?.operation &&
+        currentAlert.validator_config_json?.threshold !== undefined
+      ) {
+        setDisableSave(false);
+      } else {
+        setDisableSave(true);
+      }
+    } else {
+      setDisableSave(true);
+    }
+  };
+
+  // Initialize
+  if (
+    isEditMode &&
+    (!currentAlert ||
+      !currentAlert.id ||
+      (alert && alert.id !== currentAlert.id) ||
+      (isHidden && show))
+  ) {
+    if (alert && alert.id !== null && !loading) {
+      const id = alert.id || 0;
+
+      fetchResource(id).then(() => {
+        if (resource) {
+          setContentType(resource.chart ? 'chart' : 'dashboard');
+
+          // Add notification settings
+          const settings = (resource.recipients || []).map(setting => ({
+            method: setting.type as NotificationMethod,
+            // @ts-ignore: Type not assignable
+            recipients: (JSON.parse(setting.recipient_config_json) || {})
+              .target,
+            options: NOTIFICATION_METHODS as NotificationMethod[], // Need better logic for this
+          }));
+
+          setNotificationSettings(settings);
+
+          setCurrentAlert({
+            ...resource,
+            chart: resource.chart
+              ? getChartData(resource.chart) || { value: resource.chart.id }
+              : undefined,
+            dashboard: resource.dashboard
+              ? getDashboardData(resource.dashboard) || {
+                  value: resource.dashboard.id,
+                }
+              : undefined,
+            database: resource.database
+              ? getSourceData(resource.database) || {
+                  value: resource.database.id,
+                }
+              : undefined,
+            // log_retention: { value: resource.log_retention },
+            owners: (resource.owners || []).map(owner => ({
+              value: owner.id,
+              label: `${(owner as Owner).first_name} ${
+                (owner as Owner).last_name
+              }`,
+            })),
+            // @ts-ignore: Type not assignable
+            validator_config_json: JSON.parse(resource.validator_config_json),
```

Comment:
if you assign this to a var with a type, you should be able to assign it. Eg,

```typescript
const validatorConfig: AlertObject["validator_config_json"] = JSON.parse(resource.validator_config_json)

...

validator_config_json: validatorConfig
```

or `(JSON.parse(resource.validator_config_json) as AlertObject["validator_config_json"])`

Reply:
I think the issue here if more that `resource` is expecting `validator_config_json` to be an object, not a string. I considered adding `string` to the type definition but it ended up creating more errors/checks I would need to handle, so ignoring this one case with the stringified response felt like less work in the long run.

Reply:
right, I don't think having to always serialize the validator config object to json before adding it to an `AlertObject` is desirable or practical. This suggestion is to get rid of the `ts-ignore` comment by casting the output of the json parse to the desired type or to store it in typed var. 

https://stackoverflow.com/questions/38688822/how-to-parse-json-string-in-typescript